docs/architecture/gateway-lifecycle.md

```diff
@@ -0,0 +1,211 @@
+# Gateway Lifecycle
+
+This shows what happens when you create, update, or delete a Gateway.
+
+## Gateway Creation Flow
+
+```mermaid
+%%{init: {'theme':'dark', 'themeVariables': {'fontSize': '14px', 'darkMode': true, 'primaryColor': '#4f46e5', 'primaryTextColor': '#e5e7eb', 'primaryBorderColor': '#6b7280', 'lineColor': '#9ca3af', 'secondaryColor': '#1f2937', 'tertiaryColor': '#374151', 'background': '#111827', 'actorBkg': '#374151', 'actorBorder': '#6b7280', 'actorTextColor': '#e5e7eb', 'activationBkgColor': '#4f46e5', 'activationBorderColor': '#3730a3', 'signalColor': '#9ca3af', 'signalTextColor': '#e5e7eb', 'labelBoxBkgColor': '#1f2937', 'labelBoxBorderColor': '#6b7280', 'labelTextColor': '#e5e7eb', 'loopTextColor': '#e5e7eb', 'noteBkgColor': '#374151', 'noteBorderColor': '#6b7280', 'noteTextColor': '#e5e7eb'}}}%%
+sequenceDiagram
+    participant User
+    participant API as K8s API Server
+    participant NGF as NGF Controller
+    participant K8s as Kubernetes
+    participant Agent as NGINX Agent
+    participant NGINX
+
+    User->>API: Create Gateway resource
+    API->>NGF: Watch event: Gateway created
+    NGF->>NGF: Validate Gateway config
+    NGF->>K8s: Create NGINX Deployment
+    NGF->>K8s: Create LoadBalancer Service
+    K8s->>Agent: Start NGINX pod
+    Agent->>NGF: gRPC connect & register
+    NGF->>Agent: Send initial config
+    Agent->>NGINX: Write nginx.conf
+    Agent->>NGINX: Start NGINX process
+    Agent->>NGF: Report ready status
+    NGF->>API: Update Gateway status: Ready
+    API->>User: Gateway is ready
+```
+
+## What Happens During Creation
+
+### 1. User Creates Gateway
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: my-gateway
+  namespace: default
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+```
+
+### 2. NGF Controller Validates
+
+```text
+Validation Checks:
+├── GatewayClass exists and is valid
+├── Listener configuration is correct
+├── Required permissions are available
+├── Resource limits are acceptable
+└── No conflicting Gateways exist
+```
+
+### 3. Kubernetes Resources Created
+
+```text
+Resources Created:
+├── Deployment: nginx-gateway-{gateway-name}
+├── Service: nginx-gateway-{gateway-name}
+├── ConfigMap: nginx-config-{gateway-name}
+├── ServiceAccount: nginx-gateway-{gateway-name}
+└── Secrets: TLS certificates (if needed)
+```
```

Comment:
There are a handful of others resources we create. I don't expect them to all be listed, but I don't want a dev to be confused and think that this list is all we create.